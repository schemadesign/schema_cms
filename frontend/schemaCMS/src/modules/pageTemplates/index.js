export { PageTemplatesRoutines } from './pageTemplates.redux';

export * from './pageTemplates.selectors';
