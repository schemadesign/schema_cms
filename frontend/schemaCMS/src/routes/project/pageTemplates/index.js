export { default as PageTemplates } from './pageTemplates.container';
