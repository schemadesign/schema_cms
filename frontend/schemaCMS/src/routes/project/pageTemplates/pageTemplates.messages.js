/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'project.pageTemplates.title',
    defaultMessage: 'Page Templates',
  },
  subtitle: {
    id: 'project.pageTemplates.subtitle',
    defaultMessage: 'Library',
  },
  pageTemplate: {
    id: 'project.pageTemplates.pageTemplate',
    defaultMessage: 'Page Template',
  },
  blocksCounter: {
    id: 'project.pageTemplates.elementsCounter',
    defaultMessage: '{blocks, plural,one {# Block} other {# Blocks}}',
  },
  noData: {
    id: 'project.pageTemplates.noData',
    defaultMessage: 'No page templates available',
  },
});
