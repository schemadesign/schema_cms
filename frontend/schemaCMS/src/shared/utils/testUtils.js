import { Router } from 'react-router';
import { IntlProvider } from 'react-intl';
import { act, create } from 'react-test-renderer';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Theme } from 'schemaUI';
import PropTypes from 'prop-types';

import { DEFAULT_LOCALE, translationMessages } from '../../i18n';
import browserHistory from './history';

const ProvidersWrapper = ({ children }) => (
  <Router history={browserHistory}>
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <ThemeProvider theme={Theme.dark}>{children}</ThemeProvider>
    </IntlProvider>
  </Router>
);

ProvidersWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export const makeContextRenderer = async component => {
  let wrapper = null;

  await act(async () => {
    wrapper = await create(<ProvidersWrapper>{component}</ProvidersWrapper>);
  });

  return wrapper;
};
