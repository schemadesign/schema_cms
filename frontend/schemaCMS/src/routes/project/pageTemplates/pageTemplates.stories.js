import React from 'react';
import { storiesOf } from '@storybook/react';

import { PageTemplates } from './pageTemplates.component';
import { withTheme } from '../../../.storybook/decorators';
import { ROLES } from '../../../modules/userProfile/userProfile.constants';
import { pageTemplates } from '../../../modules/pageTemplates/pageTemplates.mocks';

export const defaultProps = {
  userRole: ROLES.ADMIN,
  pageTemplates,
  fetchPageTemplates: Function.prototype,
};

storiesOf('PageTemplates', module)
  .addDecorator(withTheme())
  .add('Default', () => <PageTemplates {...defaultProps} />);
