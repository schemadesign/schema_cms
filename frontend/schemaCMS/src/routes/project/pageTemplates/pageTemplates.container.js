import { connect } from 'react-redux';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { PageTemplates } from './pageTemplates.component';
import { PageTemplatesRoutines, selectPageTemplates } from '../../../modules/pageTemplates';
import { selectUserRole } from '../../../modules/userProfile';
import { selectProject } from '../../../modules/project';

const mapStateToProps = createStructuredSelector({
  pageTemplates: selectPageTemplates,
  userRole: selectUserRole,
  project: selectProject,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      fetchPageTemplates: promisifyRoutine(PageTemplatesRoutines.fetchPageTemplates),
      copyPageTemplate: promisifyRoutine(PageTemplatesRoutines.copyPageTemplate),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(PageTemplates);
