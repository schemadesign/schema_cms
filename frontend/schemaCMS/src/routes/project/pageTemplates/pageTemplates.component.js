import React, { Fragment, useState } from 'react';
import { useEffectOnce } from 'react-use';
import PropTypes from 'prop-types';
import { FormattedMessage, useIntl } from 'react-intl';
import { useHistory, useParams } from 'react-router';
import Helmet from 'react-helmet';

import { Container } from './pageTemplates.styles';
import messages from './pageTemplates.messages';
import { LoadingWrapper } from '../../../shared/components/loadingWrapper';
import { filterMenuOptions } from '../../../shared/utils/helpers';
import { MobileMenu } from '../../../shared/components/menu/mobileMenu';
import { getProjectMenuOptions } from '../project.constants';
import { ContextHeader } from '../../../shared/components/contextHeader';
import {
  BackArrowButton,
  NavigationContainer,
  PlusLink,
  LARGE_BUTTON_SIZE,
} from '../../../shared/components/navigation';
import { ListContainer, ListItem, ListItemTitle } from '../../../shared/components/listComponents';
import { CardHeader } from '../../../shared/components/cardHeader';
import extendedDayjs, { BASE_DATE_FORMAT } from '../../../shared/utils/extendedDayjs';
import { CounterHeader } from '../../../shared/components/counterHeader';
import {
  pageTemplatesMessage,
  libraryMessage,
  projectMessage,
  tabMessage,
  templatesMessage,
  ProjectBreadcrumbs,
} from '../../../shared/components/projectBreadcrumbs';
import { CopyButton } from '../../../shared/components/copyButton';
import reportError from '../../../shared/utils/reportError';

const getBreadcrumbsItems = project => [
  {
    path: `/project/${project.id}/`,
    span: projectMessage,
    h3: project.title,
  },
  {
    path: `/project/${project.id}/templates`,
    span: tabMessage,
    h3: templatesMessage,
  },
  {
    path: `/project/${project.id}/page-templates`,
    active: true,
    span: libraryMessage,
    h3: pageTemplatesMessage,
  },
];

const PageTemplate = ({ created, createdBy, name, id, blocks, copyPageTemplate, projectId }) => {
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);
  const whenCreated = extendedDayjs(created, BASE_DATE_FORMAT).fromNow();
  const list = [whenCreated, createdBy];
  const copyPageTemplateAction = async () => {
    try {
      setLoading(true);
      setError(false);
      await copyPageTemplate({ pageTemplateId: id, projectId });
    } catch (e) {
      reportError(e);
      setError(true);
    } finally {
      setLoading(false);
    }
  };

  const header = (
    <CardHeader
      list={list}
      icon={
        <CopyButton
          name={`pageTemplateCopyButton-${id}`}
          loading={loading}
          error={error}
          action={copyPageTemplateAction}
        />
      }
    />
  );
  const footer = <FormattedMessage {...messages.blocksCounter} values={{ blocks: blocks.length }} />;

  return (
    <ListItem headerComponent={header} footerComponent={footer}>
      <ListItemTitle id={`pageTemplateTitle-${id}`} to={`/page-template/${id}`}>
        {name}
      </ListItemTitle>
    </ListItem>
  );
};

PageTemplate.propTypes = {
  created: PropTypes.string.isRequired,
  createdBy: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  blocks: PropTypes.array.isRequired,
  projectId: PropTypes.string.isRequired,
  copyPageTemplate: PropTypes.func.isRequired,
};

export const PageTemplates = ({ fetchPageTemplates, pageTemplates, userRole, project, copyPageTemplate }) => {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
  const intl = useIntl();
  const history = useHistory();
  const { projectId } = useParams();
  const menuOptions = getProjectMenuOptions(projectId);
  const title = <FormattedMessage {...messages.title} />;
  const subtitle = <FormattedMessage {...messages.subtitle} />;
  const noData = <FormattedMessage {...messages.noData} />;

  const addTemplateUrl = `/project/${projectId}/page-templates/create`;
  const handleBackClick = () => history.push(`/project/${projectId}/templates`);

  useEffectOnce(() => {
    (async () => {
      try {
        await fetchPageTemplates({ projectId });
      } catch (e) {
        reportError(e);
        setError(e);
      } finally {
        setLoading(false);
      }
    })();
  });

  return (
    <Container>
      <Helmet title={intl.formatMessage(messages.title)} />
      <MobileMenu headerTitle={title} headerSubtitle={subtitle} options={filterMenuOptions(menuOptions, userRole)} />
      <ProjectBreadcrumbs items={getBreadcrumbsItems(project)} />
      <ContextHeader title={title} subtitle={subtitle}>
        <PlusLink id="createPageTemplate" to={addTemplateUrl} size={LARGE_BUTTON_SIZE} />
      </ContextHeader>
      <LoadingWrapper loading={loading} error={error} noDataContent={noData} noData={!pageTemplates.length}>
        <Fragment>
          <CounterHeader moveToTop copy={intl.formatMessage(messages.pageTemplate)} count={pageTemplates.length} />
          <ListContainer>
            {pageTemplates.map((page, index) => (
              <PageTemplate key={index} copyPageTemplate={copyPageTemplate} projectId={projectId} {...page} />
            ))}
          </ListContainer>
        </Fragment>
      </LoadingWrapper>
      <NavigationContainer fixed>
        <BackArrowButton id="backBtn" onClick={handleBackClick} />
        <PlusLink hideOnDesktop id="createPageTemplateMobile" to={addTemplateUrl} size={LARGE_BUTTON_SIZE} />
      </NavigationContainer>
    </Container>
  );
};

PageTemplates.propTypes = {
  userRole: PropTypes.string.isRequired,
  pageTemplates: PropTypes.array.isRequired,
  fetchPageTemplates: PropTypes.func.isRequired,
  copyPageTemplate: PropTypes.func.isRequired,
  project: PropTypes.object.isRequired,
};
