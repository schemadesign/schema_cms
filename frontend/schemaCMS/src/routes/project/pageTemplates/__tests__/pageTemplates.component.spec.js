import React from 'react';

import { PageTemplates } from '../pageTemplates.component';
import { defaultProps } from '../pageTemplates.stories';
import { makeContextRenderer } from '../../../../shared/utils/testUtils';

const mockPushHistory = jest.fn();

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useHistory: () => ({
    push: mockPushHistory,
  }),
  useParams: () => ({
    projectId: 'projectId',
  }),
}));

describe('PageTemplates: Component', () => {
  const render = props => makeContextRenderer(<PageTemplates {...defaultProps} {...props} />);

  it('should render correctly', async () => {
    const wrapper = await render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should fetch block and blocks', async () => {
    jest.spyOn(defaultProps, 'fetchPageTemplates');

    await render();

    expect(defaultProps.fetchPageTemplates).toHaveBeenCalledWith({ projectId: 'projectId' });
  });

  it('should redirect to create page template', async () => {
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'createPageTemplate' }).props.onClick();

    expect(mockPushHistory).toHaveBeenCalledWith('/project/projectId/page-templates/create');
  });

  it('should redirect to create page template on mobile', async () => {
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'createPageTemplateMobile' }).props.onClick();

    expect(mockPushHistory).toHaveBeenCalledWith('/project/projectId/page-templates/create');
  });

  it('should go back', async () => {
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'backBtn' }).props.onClick();

    expect(mockPushHistory).toHaveBeenCalledWith('/project/projectId/templates');
  });

  it('should redirect to page template', async () => {
    const wrapper = await render();
    wrapper.root.findByProps({ id: 'pageTemplateTitle-1' }).props.onClick();

    expect(mockPushHistory).toHaveBeenCalledWith('/page-template/1');
  });

  it('should call copy page template', async () => {
    jest.spyOn(defaultProps, 'copyPageTemplate');
    const wrapper = await render();
    await act(async () => {
      await wrapper.root.findByProps({ id: 'pageTemplateCopyButton-1' }).props.onClick();
    });

    expect(defaultProps.copyPageTemplate).toHaveBeenCalledWith({ projectId: 'projectId', pageTemplateId: 1 });
  });
});
