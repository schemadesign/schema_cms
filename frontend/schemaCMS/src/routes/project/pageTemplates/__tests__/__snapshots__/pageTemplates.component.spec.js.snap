// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTemplates: Component should render correctly 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="sc-AxhUy bGjTfb"
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#FFF",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": "18px",
          "width": "100%",
        }
      }
    >
      <div>
        <div
          className="sc-AxgMl fFPQpl"
        >
          <h2
            className="sc-fzplWN eQxyjC"
            id="headerTitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 100,
                "margin": 0,
              }
            }
          >
            Page Templates
          </h2>
          <h1
            className="sc-fznyAO dFXsby"
            id="headerSubtitle"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "24px",
                "fontWeight": 600,
                "letterSpacing": "-0.12px",
                "lineHight": 1.25,
                "margin": 0,
              }
            }
          >
            Library
          </h1>
        </div>
      </div>
      <button
        disabled={false}
        id="topHeaderOpenMenuBtn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "borderRadius": "48px",
            "color": "#FFF",
            "cursor": "pointer",
            "display": "inline-block",
            "fill": "#FFF",
            "fontSize": "18px",
            "margin": "0",
            "minHeight": "48px",
            "outline": "none",
            "padding": "0",
            "position": "absolute",
            "right": -6,
            "top": 6,
            "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
          }
        }
        type="button"
      >
        <svg
          style={
            Object {
              "fill": "#FFF",
              "height": "60px",
              "width": "60px",
            }
          }
          viewBox="0 0 10 60"
        >
          <path
            d="M5 37a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6zm0-10a3 3 0 110 6 3 3 0 010-6z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      id=""
      style={
        Object {
          "backgroundColor": "#000",
          "bottom": 0,
          "color": "#FFF",
          "maxWidth": null,
          "padding": "18px 20px 27px",
          "position": "absolute",
          "right": 0,
          "textAlign": "left",
          "top": 0,
          "transform": "translateX(100%)",
          "transition": "transform 400ms cubic-bezier(0.86, 0, 0.07, 1), visibility 0s linear 400ms",
          "visibility": 0,
          "width": "100%",
          "zIndex": 99999,
        }
      }
    >
      <div
        className="sc-Axmtr hUmGBz"
      >
        <div
          className="sc-AxheI dUMtsD"
        >
          <div
            className="sc-AxgMl fFPQpl"
          >
            <h2
              className="sc-fzplWN eQxyjC"
              id="headerTitle"
              style={
                Object {
                  "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": "24px",
                  "fontWeight": 100,
                  "margin": 0,
                }
              }
            >
              Page Templates
            </h2>
            <h1
              className="sc-fznyAO dFXsby"
              id="headerSubtitle"
              style={
                Object {
                  "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "fontSize": "24px",
                  "fontWeight": 600,
                  "letterSpacing": "-0.12px",
                  "lineHight": 1.25,
                  "margin": 0,
                }
              }
            >
              Library
            </h1>
          </div>
        </div>
        <div
          className="sc-AxmLO cZlyyh"
        >
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/"
              id="projectsNavBtn"
              onClick={[Function]}
            >
              Projects
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/user/"
              id="usersPageNavBtn"
              onClick={[Function]}
            >
              Users
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/projectId"
              id="projectDetailsNavBtn"
              onClick={[Function]}
            >
              Project details
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/projectId/datasource"
              id="dataSourceNavBtn"
              onClick={[Function]}
            >
              Data Sources
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/projectId/user"
              id="usersNavBtn"
              onClick={[Function]}
            >
              Project Users
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/projectId/content"
              id="contentNavBtn"
              onClick={[Function]}
            >
              Content
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/project/projectId/templates"
              id="templatesNavBtn"
              onClick={[Function]}
            >
              Templates
            </a>
          </div>
        </div>
        <div
          className="sc-AxmLO cZlyyh"
        >
          <div
            className="sc-fzozJi bEoyPT"
          >
            <a
              href="/settings"
              id="settingsNavBtn"
              onClick={[Function]}
            >
              Settings
            </a>
          </div>
          <div
            className="sc-fzozJi bEoyPT"
          >
            <div
              onClick={[Function]}
            >
              Log out
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD hiCoJg"
        >
          <a
            className="sc-fzpans TNlMA"
            href="/"
            id="aboutNavBtn"
            onClick={[Function]}
          >
            About Schema CMS
          </a>
          <a
            className="sc-fzpans TNlMA"
            href="/"
            id="apiNavBtn"
            onClick={[Function]}
          >
            API Doc
          </a>
          <a
            className="sc-fzpans TNlMA"
            href="/"
            id="repositoryNavBtn"
            onClick={[Function]}
          >
            GitHub Repository
          </a>
        </div>
      </div>
      <button
        disabled={false}
        id="topHeaderCloseMenuButton"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "none",
            "borderRadius": "48px",
            "color": "#FFF",
            "cursor": "pointer",
            "display": "inline-block",
            "fill": "#FFF",
            "fontSize": "18px",
            "margin": "0",
            "minHeight": "48px",
            "outline": "none",
            "padding": "0",
            "position": "absolute",
            "right": 0,
            "top": "6px",
            "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
          }
        }
        type="button"
      >
        <svg
          style={
            Object {
              "fill": "#FFF",
              "height": "60px",
              "width": "60px",
            }
          }
          viewBox="0 0 60 60"
        >
          <path
            d="M39.192 17.98l2.829 2.828L32.828 30l9.193 9.192-2.829 2.829L30 32.828l-9.192 9.193-2.829-2.829L27.172 30l-9.193-9.192 2.829-2.829L30 27.172l9.192-9.193z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="sc-fzoYkl fKfLDA"
  >
    <div
      style={
        Object {
          "borderBottom": "1px solid #a6a7aa",
          "display": "flex",
          "padding": "20px 0",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "color": "#a6a7aa",
            "display": "flex",
            "marginRight": "14px",
          }
        }
      >
        <a
          className="sc-fznzOf JkGvs"
          href="/project/1/"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <span
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "14px",
              }
            }
          >
            Project
          </span>
          <h3
            className="sc-fznMnq cawyWW"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "18px",
                "fontWeight": 600,
                "margin": 0,
              }
            }
            title=""
          >
            
          </h3>
        </a>
        <div
          style={
            Object {
              "color": "#a6a7aa",
              "marginLeft": "14px",
            }
          }
        >
          &gt;
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "color": "#a6a7aa",
            "display": "flex",
            "marginRight": "14px",
          }
        }
      >
        <a
          className="sc-fznzOf JkGvs"
          href="/project/1/templates"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <span
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "14px",
              }
            }
          >
            Tab
          </span>
          <h3
            className="sc-fznMnq cawyWW"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "18px",
                "fontWeight": 600,
                "margin": 0,
              }
            }
            title="Templates"
          >
            Templates
          </h3>
        </a>
        <div
          style={
            Object {
              "color": "#a6a7aa",
              "marginLeft": "14px",
            }
          }
        >
          &gt;
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "color": "#a6a7aa",
            "display": "flex",
            "marginRight": "14px",
          }
        }
      >
        <div
          className="sc-fzpkJw dOJVSr"
        >
          <span
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "14px",
              }
            }
          >
            Library
          </span>
          <h3
            className="sc-fznMnq cawyWW"
            style={
              Object {
                "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontSize": "18px",
                "fontWeight": 600,
                "margin": 0,
              }
            }
            title="Page Templates"
          >
            Page Templates
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-fznJRM fWcXoV"
  >
    <div
      className="sc-fznWqX izhnMz"
    >
      <h2
        className="sc-fzoiQi kWApJK"
        id="desktopHeaderTitle"
        style={
          Object {
            "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "24px",
            "fontWeight": 100,
            "margin": 0,
          }
        }
      >
        Page Templates
      </h2>
      <h1
        className="sc-fzqARJ euGmLc"
        id="desktopHeaderSubtitle"
        style={
          Object {
            "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "24px",
            "fontWeight": 600,
            "letterSpacing": "-0.12px",
            "lineHight": 1.25,
            "margin": 0,
          }
        }
      >
        Library
      </h1>
    </div>
    <div
      className="sc-fzqNqU hFpwJZ"
    >
      <div
        className="sc-fznxsB ikVhPc"
      >
        <button
          disabled={false}
          id="createPageTemplate"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#FFF",
              "border": "none",
              "borderRadius": "48px",
              "color": "#000",
              "cursor": "pointer",
              "display": "inline-block",
              "fill": "#FFF",
              "fontSize": "18px",
              "height": "60px",
              "margin": "0",
              "minHeight": "48px",
              "outline": "none",
              "padding": "0",
              "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
              "width": "60px",
            }
          }
        >
          <svg
            style={
              Object {
                "fill": "#000",
                "height": "60px",
                "shapeRendering": "crispEdges",
                "width": "60px",
              }
            }
            viewBox="0 0 60 60"
          >
            <path
              d="M32 15v13h13v4H32v13h-4V32H15v-4h13V15h4z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-fzoxKX dJioc"
  >
    <div
      className="sc-fzoKki fwmznp"
    />
    2 Page Templates
    <div
      className="sc-fzoKki fwmznp"
    />
  </div>
  <section
    className="sc-fznMAR dqZyIO"
  >
    <div
      className="sc-fzoyTs dGguFW"
      style={
        Object {
          "backgroundColor": "#1d1d20",
          "borderTop": "2px solid #FFF",
          "color": "#FFF",
          "display": "flex",
          "flexDirection": "column",
          "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "12px",
          "justifyContent": "space-between",
          "padding": "12px 14px 14px",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#71737e",
              "marginBottom": 14,
            }
          }
        >
          <div
            className="sc-fzqBkg fFYWEf"
          >
            <ul
              className="sc-fznWOq iILJxw"
            >
              <li
                className="sc-fzolEj kzHfOB"
                id="headerItem-0"
              >
                in 3 months
              </li>
              <li
                className="sc-fzolEj kzHfOB"
                id="headerItem-1"
              >
                owner
              </li>
            </ul>
            <div
              className="sc-fzqPZZ gXikXH"
            >
              <div
                className="sc-fzoant clXZJk"
                id="pageTemplateCopyButton-1"
                onClick={[Function]}
              >
                <svg
                  style={
                    Object {
                      "fill": "#FFF",
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M464 64h-48V48c0-26.51-21.49-48-48-48H48C21.49 0 0 21.49 0 48v320c0 26.51 21.49 48 48 48h16v48c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zM64 112v272H48c-8.837 0-16-7.163-16-16V48c0-8.837 7.163-16 16-16h320c8.837 0 16 7.163 16 16v16H112c-26.51 0-48 21.49-48 48zm416 352c0 8.837-7.163 16-16 16H112c-8.837 0-16-7.163-16-16V112c0-8.837 7.163-16 16-16h352c8.837 0 16 7.163 16 16v352z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <h1
          className="sc-fzoNJl iNXeca"
          id="pageTemplateTitle-1"
          onClick={[Function]}
          style={
            Object {
              "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              "fontSize": "24px",
              "fontWeight": 600,
              "letterSpacing": "-0.12px",
              "lineHight": 1.25,
              "margin": 0,
            }
          }
        >
          Page template
        </h1>
      </div>
      <div
        style={
          Object {
            "color": "#71737e",
            "marginBottom": 4,
            "marginTop": 10,
          }
        }
      >
        1 Block
      </div>
    </div>
    <div
      className="sc-fzoyTs dGguFW"
      style={
        Object {
          "backgroundColor": "#1d1d20",
          "borderTop": "2px solid #FFF",
          "color": "#FFF",
          "display": "flex",
          "flexDirection": "column",
          "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSize": "12px",
          "justifyContent": "space-between",
          "padding": "12px 14px 14px",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#71737e",
              "marginBottom": 14,
            }
          }
        >
          <div
            className="sc-fzqBkg fFYWEf"
          >
            <ul
              className="sc-fznWOq iILJxw"
            >
              <li
                className="sc-fzolEj kzHfOB"
                id="headerItem-0"
              >
                in 3 months
              </li>
              <li
                className="sc-fzolEj kzHfOB"
                id="headerItem-1"
              >
                owner 2
              </li>
            </ul>
            <div
              className="sc-fzqPZZ gXikXH"
            >
              <div
                className="sc-fzoant clXZJk"
                id="pageTemplateCopyButton-2"
                onClick={[Function]}
              >
                <svg
                  style={
                    Object {
                      "fill": "#FFF",
                      "height": "20px",
                      "width": "20px",
                    }
                  }
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M464 64h-48V48c0-26.51-21.49-48-48-48H48C21.49 0 0 21.49 0 48v320c0 26.51 21.49 48 48 48h16v48c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zM64 112v272H48c-8.837 0-16-7.163-16-16V48c0-8.837 7.163-16 16-16h320c8.837 0 16 7.163 16 16v16H112c-26.51 0-48 21.49-48 48zm416 352c0 8.837-7.163 16-16 16H112c-8.837 0-16-7.163-16-16V112c0-8.837 7.163-16 16-16h352c8.837 0 16 7.163 16 16v352z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <h1
          className="sc-fzoNJl iNXeca"
          id="pageTemplateTitle-2"
          onClick={[Function]}
          style={
            Object {
              "fontFamily": "Inter, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              "fontSize": "24px",
              "fontWeight": 600,
              "letterSpacing": "-0.12px",
              "lineHight": 1.25,
              "margin": 0,
            }
          }
        >
          Page template 2
        </h1>
      </div>
      <div
        style={
          Object {
            "color": "#71737e",
            "marginBottom": 4,
            "marginTop": 10,
          }
        }
      >
        1 Block
      </div>
    </div>
  </section>
  <div
    className="sc-fzqNJr fhMLFD"
  >
    <div
      className="sc-fzoLag hbFwVz"
    >
      <div
        className="sc-fzoyAV huHdwS"
      >
        <button
          disabled={false}
          id="backBtn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#1d1d20",
              "border": "none",
              "borderRadius": "48px",
              "color": "#FFF",
              "cursor": "pointer",
              "display": "inline-block",
              "fill": "#FFF",
              "fontSize": "18px",
              "height": "60px",
              "margin": "0",
              "minHeight": "48px",
              "outline": "none",
              "padding": "0",
              "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
              "width": "60px",
            }
          }
        >
          <svg
            style={
              Object {
                "fill": "#FFF",
                "height": "60px",
                "width": "60px",
              }
            }
            viewBox="0 0 60 60"
          >
            <path
              d="M25.857 17L28 19.083 18.828 28H45v3H18.828L28 39.917 25.857 42 13 29.5 25.857 17z"
            />
          </svg>
        </button>
        <div
          className="sc-fznxsB guWwxz"
        >
          <button
            disabled={false}
            id="createPageTemplateMobile"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#FFF",
                "border": "none",
                "borderRadius": "48px",
                "color": "#000",
                "cursor": "pointer",
                "display": "inline-block",
                "fill": "#FFF",
                "fontSize": "18px",
                "height": "60px",
                "margin": "0",
                "minHeight": "48px",
                "outline": "none",
                "padding": "0",
                "transition": "background 0.25s ease-in-out, color 0.25s ease-in-out",
                "width": "60px",
              }
            }
          >
            <svg
              style={
                Object {
                  "fill": "#000",
                  "height": "60px",
                  "shapeRendering": "crispEdges",
                  "width": "60px",
                }
              }
              viewBox="0 0 60 60"
            >
              <path
                d="M32 15v13h13v4H32v13h-4V32H15v-4h13V15h4z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
